aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+
+HTML Helper alternative to Input Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Html.TextBox``, ``Html.TextBoxFor``, ``Html.Editor`` and ``Html.EditorFor`` have overlapping features with the Input Tag Helper. The Input Tag Helper will automatically set the ``type`` attribute; ``Html.TextBox`` and ``Html.TextBoxFor`` will not. ``Html.Editor`` and ``Html.EditorFor`` handle collections, complex objects and templates; the Input Tag Helper does not. The Input Tag Helper, ``Html.EditorFor``  and  ``Html.TextBoxFor`` are strongly typed (they use lambda expressions); ``Html.TextBox`` and ``Html.Editor`` are not (they use expression names).
+
+Expression names 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``asp-for`` attribute value is a `ModelExpression <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/ModelExpression/index.html>`__ and the right hand side of a lambda expression. Therefore, ``asp-for="Property1"`` becomes ``m => m.Property1`` in the generated code which is why you don't need to prefix with ``Model``. You can use the "@" character to start an inline expression and move before the ``m.``:
+
+.. code-block:: HTML
+
+  @{
+      var joe = "Joe";
+  }
+  <input asp-for="@joe" />
+
+Generates the following:
+
+.. code-block:: HTML
+  
+    <input type="text" id="joe" name="joe" value="Joe" />
+
+
+Navigating child properties 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also navigate to child properties using the property path of the view model. Consider a more complex model class that contains a child ``Address`` property.
+
+.. literalinclude::  forms/sample/final/ViewModels/AddressViewModel.cs
+  :language: c#
+  :lines: 5-8
+  :dedent: 3
+  :emphasize-lines: 1-
+
+.. literalinclude::  forms/sample/final/ViewModels/RegisterAddressViewModel.cs
+  :language: c#
+  :lines: 5-11
+  :dedent: 3
+  :emphasize-lines: 6
+
+In the view, we bind to ``Address.AddressLine1``: 
+
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterAddress.cshtml 
+  :language: HTML
+  :emphasize-lines: 6
+
+The following HTML is generated for `` Address.AddressLine1``:
+
+.. code-block:: HTML
+
+  <input type="text" id="Address_AddressLine1" name="Address.AddressLine1" value="" />
+  
+Expression names and Collections
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sample, a model containing an array of ``Colors``:
```

Comment:
- Property is a *list* and it contains `string`s, not `Colors`.
- Name the property `FavoriteColors` since you're assigning them to people.
- Suggest "Sample, a model" -> "For example, with a model".